### DISCUSSION TITLE
How to configure Loco to use MongoDB as the database?

### DISCUSSION THREAD

**Discussion by magicfive2023** [loco-rs/loco#341] [General]
How to configure Loco to use MongoDB as the database? Could you provide an example project?
Thanks

**Comment by lpj145** [comment]
@magicfive2023 you can't, loco uses sea-orm under the hood, [here](https://www.sea-ql.org/SeaORM/docs/install-and-config/database-and-async-runtime/#database_driver) you can what drivers(sql servers) is supported by sea-orm.

But you can add a new "state" or extension to the app context, in this way you need to choose some rust mongodb driver, makes your own connection and pool and passes as reference to the axum router as you can see [here](https://docs.rs/axum/latest/axum/extract/struct.State.html#with-router) and [here](https://loco.rs/docs/the-app/controller/#adding-extra-state)

So bad so true, keep it in mind, loco is a opinative but dynamic approach to develop web app's.
